client/web/src/enterprise/batches/Branch.tsx

```diff
@@ -29,6 +30,7 @@ export const Branch: React.FunctionComponent<React.PropsWithChildren<BranchProps
         className={classNames('text-monospace', className)}
         as={deleted ? 'del' : undefined}
     >
+        <VisuallyHidden>{deleted && 'Deleted'} Branch:</VisuallyHidden>{' '}
```

Comment:
```suggestion
        as={deleted ? 'del' : undefined}
        aria-label={`${deleted ? 'Deleted ' : ''}Branch: `}
    >
```

Agree, let's do this. Screen readers will read the `aria-label` on the group (e.g. "Branch") which gives initial description, and then users are able to navigate further to read content of the group (e.g. "main")